From 71d67882398e9a28c268e19d5a5d66ff7632c4ed Mon Sep 17 00:00:00 2001
From: Nan Wu <wnan@google.com>
Date: Tue, 22 Jul 2025 06:59:32 -0700
Subject: [PATCH] RESTRICT AUTOMERGE: Backport Cut max duration for default
 transition to 1500

3 seconds is way too long, especially for tap jacking exploitation during animation. Cut it down to 1500 for a more tolerable experience.

Bug: 266433089
Test: manual
Flag: EXEMPT BUGFIX
Change-Id: I691b7746a8b413e956fdb75c7809904c70853552
---
 .../android/wm/shell/transition/DefaultTransitionHandler.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
index 5b6993863c5d..6ab55f6fbc37 100644
--- a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
+++ b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
@@ -126,7 +126,7 @@ import java.util.function.Consumer;
 
 /** The default handler that handles anything not already handled. */
 public class DefaultTransitionHandler implements Transitions.TransitionHandler {
-    private static final int MAX_ANIMATION_DURATION = 3000;
+    private static final int MAX_ANIMATION_DURATION = 1500;
     private static final int SIZE_CHANGE_ANIMATION_DURATION = 400;
 
     private final TransactionPool mTransactionPool;
-- 
2.52.0.rc1.455.g30608eb744-goog

